server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/DelayedAllocationIT.java

```diff
@@ -102,6 +103,10 @@ public void testDelayedAllocationTimesOut() throws Exception {
      * allocation to a very small value, it kicks the allocation of the unassigned shard
      * even though the node it was hosted on will not come back.
      */
+    @TestLogging(
+        value = "org.elasticsearch.cluster.routing.allocation.allocator:TRACE",
+        reason = "https://github.com/elastic/elasticsearch/issues/93470"
+    )
```

Comment:
According to the logs the cluster state after timeout contains:
```
routing_nodes:
-----node_id[hyOLeoguSCmau-QLOIYGwQ][V]
--------[test][0], node[hyOLeoguSCmau-QLOIYGwQ], [P], s[STARTED], a[id=opXCsb3JRL29WU_6IecysQ], failed_attempts[0]
-----node_id[9yssq3usRk2eu9ps6PdrDQ][V]
---- unassigned
--------[test][0], node[null], [R], recovery_source[peer recovery], s[UNASSIGNED], unassigned_info[[reason=NODE_LEFT], at[2023-03-06T20:13:17.102Z], delayed=false, last_node[O_-X9P4XTSm6Tq_xx5w-Rw], details[node_left [O_-X9P4XTSm6Tq_xx5w-Rw]], allocation_status[no_attempt]], failed_attempts[0]
```

Interestingly the replica shard contains `delayed=false`.
I am hoping that additional trace logging would help to narrow down the cause (balance computation or reconciliation) of the failure to initialize.